astromon/db.py

```diff
@@ -195,27 +195,31 @@ def get_table(table_name, dbfile=None):
                     raise
                 # logger.warning(f'{table_name} is not in file. Returning empty table.')
                 warnings.warn(f'{table_name} is not in file. Returning empty table.')
-                return table.Table(names=DTYPES[table_name].names, dtype=DTYPES[table_name])
+                result = table.Table(names=DTYPES[table_name].names, dtype=DTYPES[table_name])
 
         elif type(con) is tables.file.File:
             try:
                 res = con.get_node(f'/{table_name}')[:]
                 if table_name in DTYPES:
                     res = res.astype(DTYPES[table_name])
-                res = table.Table((res))
-                res.convert_bytestring_to_unicode()
-                return res
+                result = table.Table((res))
+                result.convert_bytestring_to_unicode()
             except tables.NoSuchNodeError:
                 if table_name in DTYPES:
                     # logger.warning(f'{table_name} is not in file. Returning empty table.')
                     warnings.warn(f'{table_name} is not in file. Returning empty table.')
```

Comment:
This should be an exception handled by the code below. If the h5 file really does not have an expected table then something is badly wrong.

Reply:
I agree that this is not ideal. I will add a method to explicitly create tables with the right dtype and remove this.

Reply:
I did this in a recent [commit](https://github.com/sot/astromon/pull/17/commits/0cbe0c69271dbe27949f72e7b70f50741f6b7f35). I had to change a few other things, since this changes the default behavior.